tests/foreman/ui_airgun/test_activationkey.py

```diff
@@ -66,6 +67,79 @@ def module_org():
     return entities.Organization().create()
 
 
+@tier2
+@upgrade
+def test_positive_end_to_end_crud(session, module_org):
+    """Perform end to end testing for activation key component
+
+    :id: b6b98c45-e41e-4c7a-9be4-997273b7e24d
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    name = gen_string('alpha')
+    new_name = gen_string('alpha')
+    cv = entities.ContentView(organization=module_org).create()
+    cv.publish()
+    with session:
+        session.activationkey.create({
+            'name': name,
+            'lce': {ENVIRONMENT: True},
+            'content_view': cv.name,
+        })
+        assert session.activationkey.search(name)[0]['Name'] == name
+        ak_values = session.activationkey.read(name)
+        assert ak_values['details']['name'] == name
+        assert ak_values['details']['content_view'] == cv.name
+        # Update architecture with new name
+        session.activationkey.update(name, {'details.name': new_name})
+        assert session.activationkey.search(new_name)[0]['Name'] == new_name
+        assert not session.activationkey.search(name)
+        # Delete architecture
+        session.activationkey.delete(new_name)
+        assert not session.activationkey.search(new_name)
+
+
+@tier3
+@upgrade
```

Comment:
going to remove?
```
    @run_only_on('sat')
    @stubbed()
    @tier3
    def test_positive_end_to_end(self):
```

Reply:
Should i? Docstring of that stub says about full host provisioning and auto-registering of host during provisioning flow. However it's not required to make sure AK works do i decided to write simpler and faster test with host registering via AK. But that doesn't mean original scenario doesn't have any value and it may still be a good idea to automate it (maybe not as 'end_to_end' test though).

Reply:
We need to analyze whether specific scenario will be ever automated. My expectation that we leave as less tier2/tier3 scenario behind us as we can.